libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -45,13 +48,32 @@ const renderSectionHeader = (item: { section: SectionListData<ActivityData, Sect
 
 const emptyIconSize = getCustomSize(isWeb ? 30 : 36);
 
-export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, tokenAddress = '' }) => {
+let numberOfAttempts = 0;
+
+export const ActivitySectionList: FC<Props> = ({ publicKeyHash, chainId, filterTypeName, tokenAddress = '' }) => {
   const [offsetY, setOffsetY] = useState(0);
-  const { activity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+  const { activity: allActivity, fetch, isLoading } = useAllActivity(publicKeyHash, getDebankId(chainId), tokenAddress);
+
+  const activity = useMemo(() => getFilteredActivity(allActivity, filterTypeName), [allActivity, filterTypeName]);
+
+  useEffect(() => {
+    if (
+      filterTypeName !== ActivityFilterEnum.AllActivity &&
+      allActivity.length > 0 &&
+      activity.length === 0 &&
```

Comment:
what if activity.length become 1 ? do we need more fetch request or just show 1 activity result ?